ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala

```diff
@@ -47,18 +47,17 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag, M <: SparkModel[M]
   }
 
   override def fit(dataset: Dataset[_]): M = {
-    import org.apache.spark.sql.functions.col
 
     // Update H2O params based on provided configuration
     updateH2OParams()
 
-    // if this is left empty select
-    if (getFeaturesCols().isEmpty) {
-      setFeaturesCols(dataset.columns.filter(n => n.compareToIgnoreCase(getLabelCol()) != 0))
+    if(dataset.columns.contains(getFeaturesCol())){
+      throw new IllegalArgumentException(s"Can not find features column '${getFeaturesCol()}' in the dataset")
     }
 
-    val cols = getFeaturesCols().map(col) ++ Array(col(getLabelCol()))
-    val input = hc.asH2OFrame(dataset.select(cols: _*).toDF())
+    // To find the model, we just need two columns, label & features
+    // We are sure that features column is array of doubles
+    val input = hc.asH2OFrame(dataset.select(getLabelCol(), getFeaturesCol()).toDF())
```

Comment:
Actually, we will need to convert also fold column and weight column, if set as they need to be available on the H2O frame